hal/src/nRF52840/i2c_hal.cpp

```diff
@@ -71,14 +88,20 @@ typedef enum {
     TRANSFER_STATE_ERROR_DATA
 } transfer_state_t;
 
+typedef enum hal_i2c_state_t {
```

Comment:
Considering move it to i2c_hal.h, since it can be shared with Gen 2 platforms.

Reply:
Fixed